### DISCUSSION TITLE
Remove node and connect the next node with previous node ( get error  )

### DISCUSSION THREAD

**Discussion by alikhani-dev** [xyflow/xyflow#2700] [Q&A] [answered]
Hi,
When using `useRemoveNode` in the custom node and clicking on `<button onClick={remove}>remove<button>` get an error,
I want to remove the specific node and edge and **connect the next node with prev node**  like this video : 

https://user-images.githubusercontent.com/87765316/208686644-2a4f2401-22f3-40e9-9ecf-411d34248bd0.mp4

my code :
```
const useRemoveNode = (id: NodeProps['id']) => {
	const { getNode, getNodes, getEdges, setEdges, deleteElements, setNodes } = useReactFlow()

	const onClick = useCallback(() => {
		const node = getNode(id)

		if (!node) {
			return
		}

		const nodes = getNodes()
		const edges = getEdges()
		const [prevNode] = getIncomers(node, nodes, edges)
		const [nextNode] = getOutgoers(node, nodes, edges)

		const connectedEdges = getConnectedEdges([node], edges)

		const insertEdge: Edge = {
			id: `${prevNode.id}=>${nextNode.id}`,
			target: nextNode.id,
			source: prevNode.id,
			type: ['correct', 'wrong', 'start'].includes(prevNode.type!) ? 'placeholder' : 'workflow',
			}

			deleteElements({ nodes: [node], edges: connectedEdges })
			setEdges(prev => prev.concat(insertEdge))
	
	}, [deleteElements, getEdges, getNode, getNodes, id, setEdges, setNodes])

	return onClick
}
```

console error :

![console-error](https://user-images.githubusercontent.com/87765316/208672380-3d5b0780-f60c-43e2-83c9-46cb1be1b8aa.png)
![console-error](https://user-images.githubusercontent.com/87765316/208672373-5518beb7-3870-44b2-b88a-a207a6fa6e8e.png)

**Comment by chrtze** [comment, marked as answer]
Hey @alikhani-dev, I wasn't able to reproduce this from your code snippet (please see my codesandbox here: https://codesandbox.io/s/upbeat-sunset-9ve5sw?file=/ColorSelectorNode.js). I could imagine that this has to do with the context menu. Could you provide a reproducable example?
